NuGet/NuGet.Jobs
# [Revalidation] Improve throughput by batching revalidations

Increase the throughput of the revalidation job by batching revalidations. The revalidation job does a bunch of work before enqueueing revalidations as it must:

1. Ensure the revalidation job hasn't been killswitched
2. Verify that the ingestion pipeline is healthy
3. Verify that the desired package event rate hasn't been reached

This change amortizes that work by enqueueing revalidations in batches. Also, the job now takes into account how long it spent enqueueing revalidations when deciding how long to sleep for.

Addresses https://github.com/NuGet/Engineering/issues/1877
